dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -217,11 +217,17 @@ protected void updateDBWorkflowWithSourceControlWorkflow(Workflow workflow, Work
                 }
                 workflowVersionFromDB.update(version);
             } else {
-                // create a new one and replace the old one
+                // attach real workflow
+                workflow.addWorkflowVersion(version);
+
                 final long workflowVersionId = workflowVersionDAO.create(version);
                 releaseCreated = true;
                 workflowVersionFromDB = workflowVersionDAO.findById(workflowVersionId);
                 workflow.getWorkflowVersions().add(workflowVersionFromDB);
+                // create a new one and replace the old one if it exists
+                if (existingVersionMap.containsKey(workflowVersionFromDB.getName())) {
```

Comment:
I don't understand the need for this -- on line 214 you've already verified that the map does not contain the name. How could the map now contain the name? Can line 221/223 change the name? Otherwise this seems redundant. Or I'm missing something.

Reply:
I think it should be redundant, I probably had something else in mind.